### DISCUSSION TITLE
Identify specific existing paths or objects to transform

### DISCUSSION THREAD

**Discussion by JustEricE** [spakin/SimpInkScr#90] [Q&A] [answered]
Is there a way to identify a specific existing path or object to transform or manipulate using  Simple Inkscape Scripting? I am looking to automate translations on objects I am manually creating, but the only functionality seems to be all_shapes() which can only broadly filter existing inscape objects by their type, i.e. rect, path, etc.

I am looking to select and process objects/paths based on their individual ID, title, or even "inkscape:label". Is there any way to do this?

Thank you!

**Comment by spakin** [comment, marked as answer]
Title is a little tricky because it's a nested element rather than an attribute so let me focus just on ID and Inkscape label.  The idea is to iterate over `all_shapes` and, for each object, check its `svg_get('id')` or `svg_get('inkscape:label')` to see if it's a shape in which you're interested:

```Python
for obj in all_shapes():
    if obj.svg_get('id') in ['rect123', 'path456']:
        # Do something
    if obj.svg_get('inkscape:label') in ['interesting', 'very-interesting']:
        # Do something
```

**Reply by JustEricE** [reply]
That works and I will try it! This was the thing holding me back from using simple scripting!

One other question if you don't mind. It seems that the Inset and Outset Path commands are not supported. Am I correct in assuming that? Will it be added sometime in the future? I use Inkscape to create layered geometric designs to laser cut and scaling creates less optimal results when working with paths of different sizes and shapes than Inset does.

Thanks!

**Reply by spakin** [reply]
To my knowledge, you're correct: Inset Path and Outset Path are not supported.  If there were a function in Inkscape's [`inkex` API](https://inkscape.gitlab.io/extensions/documentation/source/index.html) or a command-line Inkscape action that implemented Inset/Outset Path, Simple Inkscape Scripting definitely would support these features.
